scanpy/logging.py

```diff
@@ -1,136 +1,92 @@
 """Logging and Profiling
 """
+import logging
+import time as time_
+from logging import CRITICAL, ERROR, WARNING, INFO, DEBUG, NOTSET
+from datetime import datetime
 
-import time as time_module
-from datetime import timedelta, datetime
-from typing import Union
+import anndata.logging
 
-from anndata import logging
-from ._settings import settings
 
+HINT = (INFO + DEBUG) // 2
+logging.addLevelName(HINT, 'HINT')
 
-_VERBOSITY_LEVELS_FROM_STRINGS = {
-    'error': 0,
-    'warn': 1,
-    'info': 2,
-    'hint': 3,
-    'debug': 4,
-}
 
+class RootLogger(logging.RootLogger):
+    def __init__(self, level):
+        super().__init__(level)
+        self.propagate = False
+        self.last_time = time_.time()
 
-def info(*args, **kwargs):
-    return _msg(*args, v='info', **kwargs)
+    def log(self, level, msg, *, extra=None, deep=None, time_passed=None):
```

Comment:
Ah, you even used the `time_passed` argument. 🙂

Reply:
yeah! I heavily modified your code, but based the implementation on it.